src/api/ledger/utils.js

```diff
@@ -105,6 +105,25 @@ function hasCompleteLedgerRange(remote: Remote, minLedgerVersion?: number,
     maxLedgerVersion || remote.getLedgerSequence());
 }
 
+function hasLedger(remote: Remote, ledger: number) {
+  return remote.getServer().hasLedger(ledger);
+}
+
+function isNonContiguousLedgerRange(remote: Remote, minLedgerVersion?: number,
+                                    maxLedgerVersion?: number
+): boolean {
+  if (minLedgerVersion === undefined || maxLedgerVersion === undefined) {
+    return false;
+  }
+  return !hasCompleteLedgerRange(remote, minLedgerVersion, maxLedgerVersion)
+    && hasLedger(minLedgerVersion) && hasLedger(maxLedgerVersion);
```

Comment:
There are two `?` syntaxes, one for optional parameters, and one for maybe types. Here you need the `?` before the type name for a maybe type.